policies/module-types/template/src/lib.rs

```diff
@@ -0,0 +1,244 @@
+            }
+            _ => (),
+        }
+
+        Ok(())
+    }
+
+    fn check_apply(&mut self, mode: PolicyMode, parameters: &Parameters) -> CheckApplyResult {
+        assert!(self.validate(parameters).is_ok());
+        let p: TemplateParameters = serde_json::from_value(Value::Object(parameters.data.clone()))?;
+        let output_file = &p.path;
+        let output_file_d = output_file.display();
+
+        // Compute output
+        let output = p
+            .engine
+            .render(p.template_path.as_deref(), p.template_src, p.data)?;
+
+        let already_present = output_file.exists();
+
+        // Check if already correct
+        let mut content = String::new();
+        let already_correct = if already_present {
+            content = read_to_string(output_file)
+                .with_context(|| format!("Failed to read file {output_file_d}"))?;
+            if content == output {
+                true
+            } else {
+                rudder_debug!(
+                    "Output file {} exists but has outdated content",
+                    output_file_d
+                );
+                false
+            }
+        } else {
+            rudder_debug!("Output file {output_file_d} does not exist");
+            false
+        };
+
+        let outcome = match (already_correct, mode) {
+            (true, _) => Outcome::success(),
+            (false, PolicyMode::Audit) => {
+                if already_present {
+                    bail!("Output file {output_file_d} is present but content is not up to date")
+                } else {
+                    bail!("Output file {output_file_d} does not exist")
+                }
+            }
+            (false, PolicyMode::Enforce) => {
+                // Backup current file
+                if already_present {
+                    let backup_file = parameters
+                        .backup_dir
+                        .join(Backup::BeforeEdit.backup_file(output_file));
+                    fs::copy(output_file, &backup_file).with_context(|| {
+                        format!("Failed to write file {}", backup_file.display())
+                    })?;
+                }
+
+                // Write file
+                fs::write(output_file, output.as_bytes())
+                    .with_context(|| format!("Failed to write file {output_file_d}"))?;
+
+                let source_file = p
+                    .template_path
+                    .as_ref()
+                    .map(|p| format!(" (from {})", p.display()))
+                    .unwrap_or_default();
+
+                Outcome::repaired(repaired_report(
+                    &source_file,
+                    &output_file_d.to_string(),
+                    content,
+                    output,
+                    already_present,
+                ))
+            }
+        };
+        Ok(outcome)
+    }
+}
+
+pub fn repaired_report(
```

Comment:
the numder of parameters / numer of line is not good, this is a sign it is probably not the right abstraction